### PR TITLE
NIFI-13542: Fix missing max string length parameter usage in multiple…

### PR DESCRIPTION
**Pull request by mark-bathori**
… JSON Readers and Schema Access Strategy

# Summary

[NIFI-13542](https://issues.apache.org/jira/browse/NIFI-13542)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-extension-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/yaml/YamlTreeReader.java [hunk 1]
```diff
@@ -51,17 +49,12 @@ public class YamlTreeReader extends JsonTreeReader {
 
     @Override
     protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
-        final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
-        // Remove those properties which are not applicable for YAML
-        properties.remove(AbstractJsonRowRecordReader.MAX_STRING_LENGTH);
-        properties.remove(AbstractJsonRowRecordReader.ALLOW_COMMENTS);
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/yaml/YamlTreeReader.java:57]: As noted in the comment, these properties do not apply to the YAML Reader. Do more recent versions of Jackson enforce string length limitations for the YAML Reader?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for addressing this issue @mark-bathori.

Most of the changes look straightforward, but I noted one question on the YAML Factory. In earlier versions of Jackson, the YAML implementation did not use or enforce the string length limitations, which is why the related properties were not exposed. Is that changed in more recent versions of Jackson?

File: nifi-extension-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/yaml/YamlTreeReader.java (hunk 1 above)
Comment (mark-bathori) [reply nifi-extension-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/yaml/YamlTreeReader.java:57]: During the testing I found that Yaml is using an even lower 3mb upper limit on documents:
https://github.com/FasterXML/jackson-dataformats-text/tree/2.15/yaml#maximum-input-yaml-document-size-3-mb

Comment (exceptionfactory) [reply nifi-extension-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/yaml/YamlTreeReader.java:57]: Thanks for the background and reference @mark-bathori, that makes sense!

**Comment by arpadboda** [comment]
Thanks for digging down in this topic and fixing these issues, LGTM!

**Review by arpadboda — APPROVED** [review]

**Review by exceptionfactory — APPROVED** [review]
Thanks for the reply @mark-bathori, and thanks for the review @arpadboda! +1 merging

**Pull request closed without merging**